### PR TITLE
Doxygen: some fixes and cleanup

### PR DESCRIPTION
**Pull request by OlegHahm**
Putting some documentation into the correct group and introducing more substructure to the drivers group.

### CODE REVIEW

**Comment by OlegHahm** [comment]
To make the release shiny also from the documentation side.

**Comment by authmillenon** [comment]
Since I'm not here on the WE, I reassign. But in general looms good. (untested) half-ACK

File: sys/include/net/gnrc/gnrc_netdev2.h [hunk 1]
```diff
@@ -7,7 +7,9 @@
  */
 
 /**
+ * @defgroup  net_gnrc_netdev2   Adaption layer for GNRC on top of Netdev2
  * @ingroup   net_gnrc
+ * @brief     Provides the glue code for @ref gnrc on of @ref drivers_netdev_netdev2
```
Comment (cgundogan) [inline sys/include/net/gnrc/gnrc_netdev2.h]: the `top` is missing here

**Comment by cgundogan** [comment]
apart from my comment: changes look good. ACK

File: sys/include/net/gnrc/gnrc_netdev2.h (hunk 1 above)
Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: top?

Comment (cgundogan) [inline sys/include/net/gnrc/gnrc_netdev2.h]: you removed the `top` from `on top of`

Comment (OlegHahm) [inline sys/include/net/gnrc/gnrc_netdev2.h]: Ah, got it. Fixed and squashed directly.

**Pull request merged**